Week2/8_checkTypeOfData.js

```diff
@@ -11,19 +16,16 @@ console.log("arr is: Array");
 console.log("The value of my variable 'boo' is: " + boo);
 console.log("boo is: Boolean");
 
-function compareType() {
-    if (typeof num == typeof str)
-        console.log("'num' and 'str' are same type")
-    else if (typeof num == typeof arr)
-        console.log("'num' and 'arr' are same type")
-    else if ( typeof num == typeof boo)
-        console.log("'num' and 'boo' are same type")
-    else if ( typeof str == typeof arr)
-        console.log("'str' and 'arr' are same type")
-    else if ( typeof str == typeof boo)
-        console.log("'str' and 'boo' are same type")
-    else if ( typeof arr == typeof boo)
-        console.log("'arr' and 'boo' are same type")
-   else console.log("The variables have been assigned to a different types of values")
-}
-compareType()
+//8.5 Now compare the types of your different variables with one another.
+//8.6 Make sure to also show a message when the variables you are comparing are not the same type.
+function compareType(var1,var2) {
+    if (typeof var1 == typeof var2)
+        console.log( "this variable have the same type")
+        else console.log("The first variable assigned to %o and the second variable assigned to %o ",typeof var1 , typeof var2 )
+    }
+compareType(num,str);
```

Comment:
Ah nice! This looks very good.
Just a tip, I would always add brackets (`{` and `}`) around your `if-statements` because otherwise it is easier to make mistakes.

So then it would look like this:

```
function compareType(var1,var2) {
    if (typeof var1 == typeof var2) {
        console.log( "this variable have the same type");
     } else { 
        console.log("The first variable assigned to %o and the second variable assigned to %o ",typeof var1 , typeof var2 );
      } 
}
```